### PR TITLE
JdbcExtractor and QueryBasedExtractor to use WorkUnitState.getProp(#1065)

### PR DESCRIPTION
**Pull request by ggthename**
JdbcExtractor and QueryBasedExtractor to use WorkUnitState.getProp instead of WorkUnit.getProp. (WorkUnit and WorkUnitState #1065)

### CODE REVIEW

**Comment by lbendig** [comment]
I think we also need to adapt `MysqlExtractor#getConnectionUrl` and `SqlServerExtractor#getConnectionUrl` .

**Comment by chavdar** [comment]
@tuGithub can you have a look ? Is that related to any of your changes to Salesforce/Query Based Extractor ?

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/JdbcExtractor.java [hunk 1]
```diff
@@ -709,15 +709,15 @@ private void parseInputQuery(String query) {
    * @return JDBCSource
    */
   protected JdbcProvider createJdbcSource() {
-    String driver = this.workUnit.getProp(ConfigurationKeys.SOURCE_CONN_DRIVER);
-    String userName = this.workUnit.getProp(ConfigurationKeys.SOURCE_CONN_USERNAME);
+    String driver = this.workUnitState.getProp(ConfigurationKeys.SOURCE_CONN_DRIVER);
+    String userName = this.workUnitState.getProp(ConfigurationKeys.SOURCE_CONN_USERNAME);
     String password = PasswordManager.getInstance(this.workUnit)
```
Comment (jinhyukchang) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/JdbcExtractor.java]: Seems PasswordManager also has same problem. Could you pass workUnitState?

**Comment by jinhyukchang** [comment]
@ggthename As this PR became idle, I've forked your branch and addressed comment with new PR.
https://github.com/linkedin/gobblin/pull/1309 Your commit is still included and let me know if there's any problem. Thanks!

**Comment by jinhyukchang** [comment]
Please note that the fix has been merged in separate PR.
https://github.com/linkedin/gobblin/pull/1309

**Comment by abti** [comment]
Since the changes here are included in another PR as pointed by Jin, closing this.

**Pull request closed without merging**